worlds/noita/Rules.py

```diff
@@ -0,0 +1,132 @@
+def ban_items_from_shops(world: MultiWorld, player: int) -> None:
+    for location_name in Locations.location_name_to_id.keys():
+        if "Shop Item" not in location_name:
+            continue
+        forbid_items_at_location(world, location_name, items_hidden_from_shops, player)
+
+
+# Prevent high tier wands from appearing in early Holy Mountain shops
+def ban_early_high_tier_wands(world: MultiWorld, player: int) -> None:
+    for i, region_name in enumerate(holy_mountain_regions):
+        wands_to_forbid = wand_tiers[i+1:]
+
+        locations_in_region = Locations.location_region_mapping[region_name].keys()
+        for location_name in locations_in_region:
+            forbid_items_at_location(world, location_name, wands_to_forbid, player)
+
+    # Prevent high tier wands from appearing in the Secret shop
+    wands_to_forbid = wand_tiers[3:]
+    locations_in_region = Locations.location_region_mapping["Secret Shop"].keys()
+    for location_name in locations_in_region:
+        forbid_items_at_location(world, location_name, wands_to_forbid, player)
+
+
+def lock_holy_mountains_into_spheres(world: MultiWorld, player: int) -> None:
+    for lock in entrance_locks:
+        location = world.get_entrance(f"From {lock.source} To {lock.destination}", player)
+        GenericRules.set_rule(location, lambda state, evt=lock.event: state.has(evt, player))
+
+
+def holy_mountain_unlock_conditions(world: MultiWorld, player: int) -> None:
+    vic = world.victory_condition[player].value
+    for lock in entrance_locks:
+        if vic == Options.VictoryCondition.option_greed_ending:
+            world.get_location(lock.event, player).access_rule = \
+                lambda state, items_needed=lock.items_needed: has_perk_count(state, player, items_needed//2)
+
+        elif vic == Options.VictoryCondition.option_pure_ending:
+            world.get_location(lock.event, player).access_rule = \
+                lambda state, items_needed=lock.items_needed: has_perk_count(state, player, items_needed) \
+                and has_orb_count(state, player, items_needed)
+
+        elif vic == Options.VictoryCondition.option_peaceful_ending:
+            world.get_location(lock.event, player).access_rule = \
+                lambda state, items_needed=lock.items_needed: has_perk_count(state, player, items_needed) \
+                and has_orb_count(state, player, items_needed * 3)
+
+
+def create_all_rules(world: MultiWorld, player: int) -> None:
+    ban_items_from_shops(world, player)
+    ban_early_high_tier_wands(world, player)
+    lock_holy_mountains_into_spheres(world, player)
+    holy_mountain_unlock_conditions(world, player)
+
+    # Prevent the Map perk (used to find Toveri) from being on Toveri (boss)
+    if world.bosses_as_checks[player].value >= Options.BossesAsChecks.option_all_bosses:
+        forbid_items_at_location(world, "Toveri", {"Perk (Spatial Awareness)"}, player)

```

Comment:
You'll also want to set an access rule on the "Victory" location for the conditions that require 11 or 33 orbs. Something like so, although you can move it wherever you want.

```suggestion
        forbid_items_at_location(world, "Toveri", {"Perk (Spatial Awareness)"}, player)

    # Set rule for victory condition.
    vic = world.victory_condition[player].value
    victory_location = world.get_location("Victory", player)
    if vic == Options.VictoryCondition.option_pure_ending:
        victory_location.access_rule = lambda state: has_orb_count(state, player, 11)
    elif vic == Options.VictoryCondition.option_peaceful_ending:
        victory_location.access_rule = lambda state: has_orb_count(state, player, 33)

```

Reply:
Otherwise, calculating playthrough/win condition will not list all the required orbs for victory, only to get into each section. As it thinks you only need enough for the most expensive location.